papers/ring_vrf/zk_cont.tex

```diff
@@ -1,34 +1,101 @@
+%verifies the 
+%\end{itemize}
+%%TO DO: add an algorithm to the $\SpecialG_\rel$ such that: Our \Verify needs a separate proof-of-knowledge for $X'$, 
+%%the production of which requires knowledge of $\bar{x}$, and occurs in parallel to \Reprove.
+
+%We define (white-box) knowledge soundness for zero-knowledge continuations
+%exactly like for zero-knowledge proofs, but with the composition 
+%$\Prove : (\bar{y}, \bar{x}; \bar{\omega}) \mapsto \Reprove(\Preprove(\bar{y}, \bar{x}; \bar{\omega}))[0]$
+%as well as this additional proof-of-knowledge.
+%Zero-knowledge however should hold even if \Reprove gets invoked multiple
+%times upon the same \Preprove results, again even with the additional proof-of-knowledge.
+%\end{definition} 
+
+\begin{definition}[ZK Continuations] A zero-knowledge continuation $\ZKCont$ for a relation $\relone$ with 
+input $(\bary, \barx)$ and witness $\baromega$ is a tuple of efficient algorithms 
+($\ZKCont.\Setup$, $\ZKCont.\Gen$, $\ZKCont.\Preprove$, $\ZKCont.\Reprove$, $\ZKCont.\Verify$) 
+such that for implicit security parameter $\lambda$,
 \begin{itemize}
-\item $\SpecialG_\rel.\Preprove : (\bar{y}, \bar{x}; \bar{\omega}) \mapsto (X,\pi)$ \,
- constructs a pre-commitment $X$ and a pre-proof $\pi$ for $\rel$ from 
- transparent inputs $\bar{y}$, opaque inputs $\bar{x}$, and witnesses $\bar{\omega}$.
-\item $\SpecialG_\rel.\Reprove : (X,\pi) \mapsto ((X',\pi'); b)$ \,
- finalizes the commitment $X'$ and proof $\pi'$ and returns an opening $b$ for the commitment. 
-\item $\SpecialG_\rel.\Verify(\bar{y}; (X',\pi') )$ \, 
- verifies the 
-\end{itemize}
-Our \Verify needs a separate proof-of-knowledge for $X'$, the production
-of which requires knowledge of $\bar{x}$, and occurs in parallel to \Reprove.
 
-We define (white-box) knowledge soundness for zero-knowledge continuations
-exactly like for zero-knowledge proofs, but with the composition 
-$\Prove : (\bar{y}, \bar{x}; \bar{\omega}) \mapsto \Reprove(\Preprove(\bar{y}, \bar{x}; \bar{\omega}))[0]$
-as well as this additional proof-of-knowledge.
-Zero-knowledge however should hold even if \Reprove gets invoked multiple
-times upon the same \Preprove results,
- again even with the additional proof-of-knowledge.
+\item $\ZKCont.\Setup: (1^{\lambda}) \mapsto \crs$ a setup algorithm that on input the security parameter 
+outputs a common reference string $\crs$, 
+
+\item $\ZKCont.\Gen: (\crs, \relone) \mapsto (\crspk, \crsvk)$ \, 
+outputs a pair of proving key $\crspk$ and verification key $\crsvk$, 
+
+\item $\ZKCont.\Preprove: (\crspk, \bar{y}, \bar{x}, \bar{\omega}, \relone) \mapsto (X, \pi, b)$ \,
+constructs commitment $X$ from a vector of inputs $\bar{x}$ (called \emph{opaque}) and 
+constructs proof $\pi$ from vector of inputs 
+$\bar{y}$ (called \emph{transparent}), from $\bar{x}$ and vector of witnesses $\bar{\omega}$, and 
+also outputs $b$ as the opening for $X$,
+
+\item $\ZKCont.\Reprove: (\crspk, X, \pi, \relone) \mapsto (X', \pi', b')$ \,
```

Comment:
does it need b as an input? If not, why does preprove output it?

Reply:
Agreed, b should be an input to reprove.